helix-core/src/main/java/org/apache/helix/HelixManager.java

```diff
@@ -419,6 +420,17 @@ void addExternalViewChangeListener(org.apache.helix.ExternalViewChangeListener l
    */
   Long getSessionStartTime();
 
+  /**
+   * Checks whether the cluster manager is leader and returns the session ID associated to the
+   * connection of cluster data store, if and only if it is leader.
+   *
+   * @return {@code Optional<String>} session ID is present inside the {@code Optional} object
+   * if the cluster manager is leader. Otherwise, returns an empty {@code Optional} object.
+   */
+  default Optional<String> getSessionIdIfLead() {
```

Comment:
@jiajunwang Thanks for sharing your thought. I think the concept "fake constant session" is like an empty `Optional` object? With `Optional`, it looks more straightforward and clear than a fake constant session.
With a fake constant, we still need an `if` to check leadership for those storage systems that have leadership. I would say `null` is kind of this fake constant: if no session concept, then return null. And I think `Optional` is better than `null` here.

Reply:
Optional is not looking good here.
It might be overkill, but how about we define a "private" SessionAwareHelixManager interface? Which extends HelixManager but add getSessionIdIfLead() method.

Reply:
I thought about the private interface. My thought is that this private interface will increase more work to identify which places will need this SessionAwareHelixManager and construct it. And it doesn't converge. Eg. if there is only one place that uses the sessionAwareHelixManager and only one place checking the session, it's fine.
But if there is mixed usage, we may need to use `instaceof` to check, which doesn't look nice.